src/routes/amt/userConsent/sendUserConsentCode.ts

```diff
@@ -0,0 +1,29 @@
+/*********************************************************************
+ * Copyright (c) Intel Corporation 2021
+ * SPDX-License-Identifier: Apache-2.0
+ **********************************************************************/
+import { Response, Request } from 'express'
+import { logger as log } from '../../../utils/logger'
+import { ErrorResponse } from '../../../utils/amtHelper'
+import { MqttProvider } from '../../../utils/mqttProvider'
+export async function sendUserConsentCode (req: Request, res: Response): Promise<void> {
+  const userConsentCode = req.body.consentCode
+  const guid = req.params.guid
+  try {
+    // Cancel a previous opt-in code request.
+    req.amtStack.IPS_OptInService_SendOptInCode(userConsentCode, (stack, name, response, status) => {
+      if (status === 200 && response?.Body?.ReturnValue === 0) {
+        MqttProvider.publishEvent('success', ['Send_User_Consent_Code'], 'Sent user consent code', guid)
+        res.status(200).json(response).end()
+      } else {
+        log.error(`Fail to send user consent code for guid : ${guid}.`)
```

Comment:
if i enter a wrong code, i see this message. But i'm guessing it didnt fail to send, its just incorrect. I think we should make that distinctions and send back the right error, something like invalid code, please try again, and re-prompt the dialog box.

Reply:
Additionally, after the wrong code, and i try again it gives me an error, i would have expected a new prompt.